pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -398,6 +398,15 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 			for name, value := range annotation {
 				service.Annotations[name] = value
 			}
+
+			// Set the Load Balancer IP for external load balancers on OpenStack only
+			if platform.Type == configv1.OpenStackPlatformType {
+				if lb != nil && lb.ProviderParameters != nil &&
+					lb.ProviderParameters.Type == operatorv1.OpenStackLoadBalancerProvider &&
+					lb.ProviderParameters.OpenStack != nil {
+					service.Spec.LoadBalancerIP = lb.ProviderParameters.OpenStack.LoadBalancerIP
+				}
+			}
```

Comment:
The API definition doesn't mention that `loadBalancerIP` is only valid for external scope: https://github.com/openshift/cluster-ingress-operator/blob/0e0f3e3f26bfe8bc3737999886c20e3afd1b3c9c/vendor/github.com/openshift/api/operator/v1/types_ingress.go#L681-L683

Can you add that limitation to the API field godoc?  I expect we could even add some CEL validation rules to prevent specifying `loadBalancerIP` with `scope: Internal`.  

Alternatively, if that limitation doesn't in fact exist, then the `if` block that you are adding here should be moved outside the `if isInternal`/`else`.  

Actually, even if using `loadBalancerIP` is limited to external scope, I think the code would be easier to read if you moved the OpenStack logic to a new `case` in the `switch` below.  The `if isInternal`/`else` blocks are really more for setting the platform-specific scope annotations.

Reply:
@mdbooth do you think we should restrict this to the external scope?
I'm trying to find a reason why a FIP would be attached to a internally scoped LB.

Reply:
actually, reading CPO makes me think that this is only for external LBs: https://github.com/kubernetes/cloud-provider-openstack/blob/333a126f63038d52fad1e306eff6ee4ef7f53948/pkg/openstack/loadbalancer.go#L624-L625

Reply:
@Miciah I'm going to update the API as requested: https://github.com/openshift/api/pull/2051
And move that code into the switch later. I confirm we only want the floating IP to be for external LB.